sherlockml/clients/report.py

```diff
@@ -0,0 +1,174 @@
+        "created_at",
+        "name",
+        "id",
+        "description",
+        "active_version_id",
+        "versions",
+    ],
+)
+
+ReportVersion = namedtuple(
+    "ReportVersion",
+    [
+        "created_at",
+        "author_id",
+        "report_path",
+        "report_key",
+        "report_bucket",
+        "notebook_path",
+        "id",
+        "report_id",
+    ],
+)
+
+
+class ReportVersionSchema(Schema):
+    def __init__(self, report_id=None, *args, **kwargs):
+        super(ReportVersionSchema, self).__init__(*args, **kwargs)
+        if report_id is not None:
+            self.context["report_id"] = report_id
+
+    id = fields.UUID(data_key="version_id", required=True)
+    created_at = fields.DateTime(required=True)
+    author_id = fields.UUID(required=True)
+    report_path = fields.String(required=True)
+    report_key = fields.String(required=True)
+    report_bucket = fields.String(required=True)
+    notebook_path = fields.String(required=True)
+    report_id = fields.UUID()
+
+    @pre_load
+    def set_report_id(self, data):
+        if data.get("report_id") is None and "report_id" in self.context:
+            data["report_id"] = self.context["report_id"]
+        return data
+
+    @post_load
+    def make_report_version(self, data):
+        return ReportVersion(**data)
+
+
+class ReportSchema(Schema):
+    created_at = fields.DateTime(required=True)
+    name = fields.String(required=True, data_key="report_name")
+    id = fields.UUID(required=True, data_key="report_id")
+    description = fields.String(required=True)
+    active_version = fields.Nested(ReportVersionSchema, required=True)
+
+    @pre_load
+    def add_report_id_to_context(self, data):
+        if "report_id" in data:
+            self.context["report_id"] = data["report_id"]
+        return data
+
+    @post_load
+    def make_active_report(self, data):
+        return Report(**data)
+
+
+class VersionedReportSchema(Schema):
+
+    created_at = fields.DateTime(required=True)
+    name = fields.String(required=True, data_key="report_name")
+    id = fields.UUID(required=True, data_key="report_id")
+    description = fields.String(required=True)
+    active_version_id = fields.UUID(required=True)
+    versions = fields.Nested(ReportVersionSchema, required=True, many=True)
+
+    @pre_load
+    def add_report_id_to_context(self, data):
+        if "report_id" in data:
+            self.context["report_id"] = data["report_id"]
+        return data
+
+    @post_load
+    def make_versioned_report(self, data):
+        return VersionedReport(**data)
+
+
+def _relative_project_path(path):
+    if path.startswith("/project/"):
+        return path[8:]
+    else:
+        return path
+
+
+class ReportClient(BaseClient):
+
+    SERVICE_NAME = "tavern"
+
+    def list(self, project_id):
+        endpoint = "/project/{}".format(project_id)
+        return self._get(endpoint, ReportSchema(many=True))
+
+    def list_report_versions(self, report_id):
```

Comment:
Maybe `get_with_versions` instead? I expect to get a list of something from a method labelled `list...`, whereas here we're getting a 'VersionedReport' back.